modules/tests/shared/src/test/scala/io/circe/DecoderSuite.scala

```diff
@@ -343,4 +352,15 @@ class DecoderSuite extends CirceSuite with LargeNumberDecoderTests with TableDri
 
     assert(history.size === size + 1)
   }
+
+  case class NotDecodable(a: Int)
+  implicit val decodeNotDecodable: Decoder[NotDecodable] = Decoder.failedWithMessage("Some message")
+  implicit val encodeNotDecodable: Encoder[NotDecodable] = deriveEncoder[NotDecodable]
+  implicit val arbNotDecodable: Arbitrary[NotDecodable] = Arbitrary(Gen.resultOf(NotDecodable))
```

Comment:
To be honest I'd probably skip the encoding and arbitrary machinery here and try to decode a hard-coded `[{ "a": 1 }}`, since the goal is really just to make sure the error survives.